src/ai/backend/manager/models/rbac_models/user_role.py

```diff
@@ -41,12 +42,12 @@ class UserRoleRow(Base):
         "granted_at", sa.DateTime(timezone=True), nullable=False, server_default=sa.func.now()
     )
 
-    role_row: RoleRow = relationship(
+    role_row: Optional[RoleRow] = relationship(
         "RoleRow",
         back_populates="mapped_user_role_rows",
         primaryjoin="RoleRow.id == foreign(UserRoleRow.role_id)",
     )
-    user_row: UserRow = relationship(
+    user_row: Optional[UserRow] = relationship(
         "UserRow",
         back_populates="role_assignments",
         primaryjoin="UserRow.uuid == foreign(UserRoleRow.user_id)",
```

Comment:
I'm curious, doesn't it simply join with `RoleRow.id == UserRoleRow.role_id`?

Reply:
Do you mean a mapping between roles and user_roles? or between user_roles and users? user can be null if the user record is deleted but user role remains